optional_plugins/golang/avocado_golang/__init__.py

```diff
@@ -57,6 +61,9 @@ def test(self):
         """
         Create the Golang command and execute it.
         """
+        if _GO_BIN is None:
+            raise exceptions.TestError("go binary not be found")
+
```

Comment:
Yes, I like this solution better, but the request I got during code review was to disable the plugin from the `avocado pluigns` list when `go` is not installed. @ldoktor, could you please review this PR?

Reply:
Well the question is which way do we want to go. As `avocado-vt` disables the plugin on missing dependencies I recommended doing the same here.

Anyway if we decide not to disable the plugin, I'd recommend tweaking the loader to return on missing `go` binary nothing, or single NotGoTest along with notice about missing `go` library in verbose mode, because this way you might get unexpected result (imagine `avocado run --loaders go file -- sort` where `sort` exists in `GOPATH` as well as binary `sort` exists in `.`. Currently and with what I proposed you get SIMPLE test. With this patch you get GO error test, which is IMO not correct)

Reply:
@ldoktor I reckon that Avocado-VT is doing so because the requirements are much more complex and too many of them.  `vt-bootstrap` is a proof of the number and complexity of Avocado-VT's requirements.  Let's not forget that Avocado-VT is a separate project and requires this extra set of setup/bootstrapping.

For a plugin that is part of the same repo, we end up with extra problems if we let the plugin error out.  Avocado won't *disable* a plugin on such a failure.  For instance, running `make check` in a freshly checked out Avocado source tree will fail if `go` is not installed.  Also, we are having problems with readthedocs.org not being able to build the docs.

I don't think we have to promise or intend symmetry between listing and running tests.  There are way too many things that may be necessary for running tests that we can not predict in advance.  ERRORing a test, is IMO, the most straightforward way to do so.

Anyway, there's a typo in the given message `s/be//`, so I'll send a v2.